src/compile/ast.rs

```diff
@@ -1016,6 +1016,7 @@ pub struct MatchPattern<T = ()> {
 #[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
 pub enum MatchPatternKind<T> {
     Simple(StringId),
+    Assign(StringId),
```

Comment:
Is there a better name for the non-assignment one than `Simple`?  That made sense back when it was the only non-tuple kind, but another name might be clearer now.

Reply:
I have renamed this to `Bind`.